internal/core/processor/universal/updater/updater_test.go

```diff
@@ -0,0 +1,203 @@
+package updater_test
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/freeverseio/laos-universal-node/internal/platform/model"
+	"github.com/freeverseio/laos-universal-node/internal/platform/scan"
+
+	uUpdater "github.com/freeverseio/laos-universal-node/internal/core/processor/universal/updater"
+	mockScan "github.com/freeverseio/laos-universal-node/internal/platform/scan/mock"
+	mockTx "github.com/freeverseio/laos-universal-node/internal/platform/state/mock"
+	"github.com/freeverseio/laos-universal-node/internal/platform/state/tree/enumerated"
+	"github.com/freeverseio/laos-universal-node/internal/platform/state/tree/enumeratedtotal"
+	"github.com/freeverseio/laos-universal-node/internal/platform/state/tree/ownership"
+	"go.uber.org/mock/gomock"
+)
+
+func TestUpdateState(t *testing.T) {
```

Comment:
We need error test cases here, this function is important. If you don't have time today I can try to do it next week or we'll tackle this with a task

Reply:
yes, this is just a happy path for updater. I think this is going to change very soon so I did not insisted much writting tests. I could write some wb tests in updater but I think updater with the fixes we plan for optmization is going to be changed very soon. so I did not bother adding tests now to be honest.

Reply:
actually maybe the happy path is enough. the rest are just mocking calls that don't have much logic between